core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoRepository.java

```diff
@@ -424,9 +430,14 @@ private SslData createSslData(
           sslContext = SSLContext.getInstance("TLSv1.2");
         }
       }
-
       sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), new SecureRandom());
 
+      // If this is CAS-managed, then use the HTTPS algorithm to check the subjectAlternateNames
+      // against the certificate.
```

Comment:
should we not also be doing this for PSC? It also has the DNS name as a SAN entry...

Reply:
Actually, this is wrong. I'm removing it. It won't work because unlike Go we cannot override the hostname passed to Socket.connect(). 

I built custom hostname verification logic in `InstanceCheckingTrustManager`. 

The built-in HTTPS hostname verification algorithm won't work because the connector cannot override the hostname when `Socket.connect(addr)` is called with an IP Address.